src/katgpucbf/xbgpu/bsend.py

```diff
@@ -327,13 +326,16 @@ def enable_substream(self, stream_id: int, enable: bool = True) -> None:
     async def get_free_chunk(self) -> Chunk:
         """Obtain a :class:`.Chunk` for transmission.
 
-        If the chunk's :attr:`future` is cancelled
+        Raises
+        ------
+        asyncio.CancelledError
+            If the chunk's send future is cancelled.
         """
         chunk = await self._chunks_queue.get()
         try:
             await chunk.future
         except asyncio.CancelledError:
-            pass
+            raise asyncio.CancelledError("Sending chunk was cancelled")
```

Comment:
You can just do `raise` and it will send the exception on upwards unchanged.

Reply:
Too right, I sometimes forget "how useful" the tracebacks are when an Exception is raised. Happy to remove the ~custom message if you'd prefer it was more straightforward.

Reply:
Yeah I'd just have used a plain `raise`.
If the behaviour that you are after is raising this particular exception and catching and logging all other kinds.

Reply:
I agree
```suggestion
            raise
```